File: placementprepration/stockspan.java
import java.util.*;
public class stockspan{
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int []price=new int[n];
        int [] span=new int [n];
        for(int i=0;i<n;i++){
            price[i]=sc.nextInt();
        }
        for(int i=0;i<n;i++){
            span[i]=1;
            for(int j=i-1;j>=0;j--){
                if(price[j]<=price[i]){
                    span[i]++;
                }
                else{
                    break;
                }
            }
        }
        for(int i=0;i<n;i++){
            System.out.print(span[i]+" ");
        }
        

    }
}

File: placementprepration/stockspan.py
import sys

def stock_span(prices):
    stack = []
    spans = []
    for price in prices:
        span = 1
        while stack and stack[-1][0] <= price:
            span += stack.pop()[1]
        stack.append((price, span))
        spans.append(span)
    return spans


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if not data:
        return
    n, *rest = data
    prices = rest[:n]
    res = stock_span(prices)
    print(" ".join(map(str, res)))


if __name__ == "__main__":
    main()
